internal/autocert/manager.go

```diff
@@ -155,6 +158,7 @@ func (mgr *Manager) getCertMagicConfig(ctx context.Context, cfg *config.Config)
 		}
 	}
 	acmeMgr := certmagic.NewACMEIssuer(mgr.certmagic, mgr.acmeTemplate)
+	acmeMgr.DisableHTTPChallenge = cfg.Options.HTTPRedirectAddr == ""
```

Comment:
I think we might even want to strengthen this condition, and require that the redirect address uses port 80, as the http-01 challenge can only be done on port 80.

(see the [Let's Encrypt docs](https://letsencrypt.org/docs/challenge-types/#:~:text=The%20HTTP%2D01%20challenge%20can%20only%20be%20done%20on%20port%2080.%20Allowing%20clients%20to%20specify%20arbitrary%20ports%20would%20make%20the%20challenge%20less%20secure%2C%20and%20so%20it%20is%20not%20allowed%20by%20the%20ACME%20standard.))